// The Firmament project
// Copyright (c) 2015-2015 Ionel Gog <ionel.gog@cl.cam.ac.uk>
//
// Simulator event descriptor protobuf.

syntax = "proto3";

package firmament;

message EventDescriptor {
  enum EventType {
    ADD_MACHINE = 0;
    REMOVE_MACHINE = 1;
    UPDATE_MACHINE = 2;
    TASK_END_RUNTIME = 3;
    TASK_SUBMIT = 4;
    MACHINE_HEARTBEAT = 5;
  }

  EventType type = 1;
  uint64 machine_id = 2;
  uint64 task_index = 3;
  uint64 job_id = 4;
  float requested_cpu_cores = 5;
  uint64 requested_ram = 6; // in MB
  uint32 priority = 7;
  uint32 scheduling_class = 8;
}
